pkgs/development/libraries/http-parser/default.nix

```diff
@@ -12,7 +12,12 @@ stdenv.mkDerivation rec {
   };
 
   NIX_CFLAGS_COMPILE = "-Wno-error";
-  patches = [ ./build-shared.patch ];
+  patches = [ ./build-shared.patch ] ++ lib.optionals stdenv.isAarch32
```

Comment:
This patch applies to all platforms and doesn't need to be conditional. Since the patch removes an assertion on non-x86 platforms, we want to apply it to all of those platforms.

Reply:
Resolved with 6b88484.

Reply:
I agree in principle, and prefer it to be done this way.

Using conditionals is usually done to "work around" needing to go through staging. This is why you'll see it around armv7l fixes, or other platform-specific which actually should just be applied outright.

This is why I didn't speak about it, it's sadly the "preferred" way to do things.

Though, with that said, one benefit there is to apply a patch unconditionally is that when it stops working, it stops working early.